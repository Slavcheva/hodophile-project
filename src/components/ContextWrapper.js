import React, { createContext } from 'react'

export const AuthContext = createContext({isLogged: false })

const ContextWrapper = props => {
  const [isLogged, setIsLogged] = React.useState(false)

  function parseCookies() {
    return document.cookie.split('; ').reduce((acc, cookie) => {
      const [cookieName, cookieValue] = cookie.split('=');
      acc[cookieName] = cookieValue;
      return acc;
    }, {})
  }

  React.useEffect(()=>{

    const cookies = parseCookies();
    const isLogged = !!cookies['x-auth-token'];

    setIsLogged(isLogged)
  })
  return (
    <AuthContext.Provider value={{isLogged}}>
      {props.children}
    </AuthContext.Provider>
  )
}

export default ContextWrapper