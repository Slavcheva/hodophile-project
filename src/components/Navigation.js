import React from 'react'
import {Link} from "react-router-dom";


function Navigation() {
    return <div>

        <nav className="nav">
            <div className="container">
                <div className="logo">
                    <Link to="/">
                        <p>Logo</p>
                        {/*<img src={logo} alt="header-logo" className="header-logo"/>*/}
                    </Link>
                </div>
                <div className="main_list">
                    <ul className="nav-links">
                        <Link to="/login" className="header-nav-link">Sign in</Link>
                        <Link to="/register" className="header-nav-link">Sign up</Link>
                        <Link to="/profile" className="header-nav-link">Profile</Link>
                    </ul>
                </div>
                <br/>
                <div className="main_list">
                    <ul className="nav-links">
                        <Link to="/trips" className="nav-link">Пътеписи</Link>
                        <Link to="/destinations" className="nav-link">Дестинации</Link>
                        <Link to="/create-trip" className="nav-link">Създайте пътепис</Link>
                    </ul>
                </div>
            </div>
        </nav>

        {/*<section className="home">*/}
        {/*</section>*/}
        {/*    <div>*/}
        {/*        <p className="myP">*/}
        {/*            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Ipsum ratione facere animi impedit rem labore sint repellendus ipsa sapiente voluptatem aut excepturi quo itaque, ab earum cumque. Voluptatem beatae id inventore quod voluptate qui deserunt, quis placeat, tempora ex totam, dolore sequi harum eos voluptatibus animi labore officiis minus laboriosam*/}
        {/*            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Ipsum ratione facere animi impedit rem labore sint repellendus ipsa sapiente voluptatem aut excepturi quo itaque, ab earum cumque. Voluptatem beatae id inventore quod voluptate qui deserunt, quis placeat, tempora ex totam, dolore sequi harum eos voluptatibus animi labore officiis minus laboriosam*/}
        {/*        </p>*/}
        {/*    </div>*/}
    </div>
}

export default Navigation
