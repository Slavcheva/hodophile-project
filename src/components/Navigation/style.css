nav img {
    max-width: 80px;
    height: auto;
    margin: 0 auto;
}
nav.navbar-menu {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    background-color: #A678CF;
}
.nav-links, h2{
    margin: 5px;
}
.nav-links a {
    /*display: flex;*/
    border-radius: 0.5em;
    justify-content: space-evenly;
    text-decoration: none;
    color: black;
    width: 15%;
    align-items: center;
    padding: 10px 10px;
    border: 1px solid #A678CF;
    font-size: 35px;

}

.nav-links a:hover {
    color: #A678CF;
    background-color: white;
}