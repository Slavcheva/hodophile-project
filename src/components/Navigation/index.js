import React from 'react'
import {Link} from "react-router-dom";

import logo from '../../icons8-suitcase-100.png'
import './style.css'
import {AuthContext} from "../ContextWrapper";

function Navigation({userD}) {
    const name = userD.username

    return <nav className="navbar-menu">
        <div className="logo">
            <Link to="/">
                <img src={logo} alt="header-logo" className="header-logo"/>
            </Link>
        </div>
        <h2>Hodophile - "One who loves to travel"</h2>
        <AuthContext.Consumer>
            {
                value => (<div className="navbar">
                    <ul className="nav-links">
                        {
                            value.isLogged
                                ? <span>Hello, {name}! </span>
                                : undefined
                        }
                        {!value.isLogged && <Link to="/login" className="header-nav-link">Sign in</Link>}
                        {!value.isLogged && <Link to="/register" className="header-nav-link">Sign up</Link>}
                        {value.isLogged && <Link to="/logout" className="header-nav-link">Sign out</Link>}
                    </ul>
                </div>)
            }
        </AuthContext.Consumer>
    </nav>
}

export default Navigation
