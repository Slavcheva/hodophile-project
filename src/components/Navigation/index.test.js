import React from "react";
import renderer from 'react-test-renderer'
import {AuthContext} from '../ContextWrapper'
import Navigation from "./index";

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('Navigation component', () => {
    test('should render correctly if user is is logged', () => {
        const component = renderer.create(
            <AuthContext.Provider value={{isLogged: true}}>
                <Navigation userD={{username:'Pesho'}}/>
            </AuthContext.Provider>
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('should render correctly if user is not is logged', () => {
        const component = renderer.create(
            <AuthContext.Provider value={{isLogged: false}}>
                <Navigation userD={{}}/>
            </AuthContext.Provider>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})