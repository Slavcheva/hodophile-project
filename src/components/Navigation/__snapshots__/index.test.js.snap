// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render correctly if user is is logged 1`] = `
<nav
  className="navbar-menu"
>
  <div
    className="logo"
  >
    <Link
      to="/"
    >
      <img
        alt="header-logo"
        className="header-logo"
        src="icons8-suitcase-100.png"
      />
    </Link>
  </div>
  <h2>
    Hodophile - "One who loves to travel"
  </h2>
  <div
    className="navbar"
  >
    <ul
      className="nav-links"
    >
      <span>
        Hello, 
        Pesho
        ! 
      </span>
      <Link
        className="header-nav-link"
        to="/logout"
      >
        Sign out
      </Link>
    </ul>
  </div>
</nav>
`;

exports[`Navigation component should render correctly if user is not is logged 1`] = `
<nav
  className="navbar-menu"
>
  <div
    className="logo"
  >
    <Link
      to="/"
    >
      <img
        alt="header-logo"
        className="header-logo"
        src="icons8-suitcase-100.png"
      />
    </Link>
  </div>
  <h2>
    Hodophile - "One who loves to travel"
  </h2>
  <div
    className="navbar"
  >
    <ul
      className="nav-links"
    >
      <Link
        className="header-nav-link"
        to="/login"
      >
        Sign in
      </Link>
      <Link
        className="header-nav-link"
        to="/register"
      >
        Sign up
      </Link>
    </ul>
  </div>
</nav>
`;
